hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/ITestReadBufferManager.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.hadoop.fs.azurebfs.AzureBlobFileSystem;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.*;
+
+import static org.apache.hadoop.fs.azurebfs.constants.ConfigurationKeys.*;
+import static org.apache.hadoop.fs.azurebfs.constants.FileSystemConfigurations.*;
+
+public class ITestReadBufferManager extends AbstractAbfsIntegrationTest {
+
+    public ITestReadBufferManager() throws Exception {
+    }
+
+    @Test
+    public void testPurgeBufferManagerForParallelStreams() throws Exception {
+        describe("Testing purging of buffers from ReadBufferManager for " +
+                "parallel input streams");
+        final int numBuffers = 16;
+        final LinkedList<Integer> freeList = new LinkedList<>();
+        for (int i=0; i < numBuffers; i++) {
+            freeList.add(i);
+        }
+        AzureBlobFileSystem fs = getABFSWithReadAheadConfig();
+        for(int i=0; i < 4; i++) {
+            String fileName = methodName.getMethodName() + i;
+            byte[] fileContent = getRandomBytesArray(ONE_MB);
+            Path testFilePath = createFileWithContent(fs, fileName, fileContent);
+            try (FSDataInputStream iStream = fs.open(testFilePath)) {
+                iStream.read();
+            }
+        }
+        ReadBufferManager bufferManager = ReadBufferManager.getBufferManager();
+        Assertions.assertThat(bufferManager.getCompletedReadListCopy().size())
+                .describedAs("After closing all streams completed list size should be 0")
+                .isEqualTo(0);
+
+        Assertions.assertThat(bufferManager.getInProgressCopiedList().size())
+                .describedAs("After closing all streams inProgress list size should be 0")
+                .isEqualTo(0);
+        Assertions.assertThat(bufferManager.getFreeListCopy())
+                .describedAs("After closing all streams free list contents should match with " + freeList)
+                .hasSize(numBuffers)
+                .containsExactlyInAnyOrderElementsOf(freeList);
+        Assertions.assertThat(bufferManager.getReadAheadQueueCopy())
+                .describedAs("After closing all stream ReadAheadQueue should be empty")
+                .hasSize(0);
+
+    }
+
+    @Test
+    public void testPurgeBufferManagerForSequentialStream() throws Exception {
+        describe("Testing purging of buffers in ReadBufferManager for " +
+                "sequential input streams");
+        AzureBlobFileSystem fs = getABFSWithReadAheadConfig();
+        final String fileName = methodName.getMethodName();
+        byte[] fileContent = getRandomBytesArray(ONE_MB);
+        Path testFilePath = createFileWithContent(fs, fileName, fileContent);
+        AbfsInputStream iStream1 = (AbfsInputStream) fs.open(testFilePath).getWrappedStream();
+        iStream1.read();
+        // closing the stream right away.
+        iStream1.close();
```

Comment:
can you close these always in a try/finally, just so if an assert fails things still get cleaned up